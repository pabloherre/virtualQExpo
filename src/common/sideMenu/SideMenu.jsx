import React from 'react';
import { connect } from 'react-redux';
import RoundedButton from '../buttons/RoundedButton/RoundedButton';
import { View } from 'react-native';
import AuthService from '../../modules/auth/Auth.service';
import UserService from '../../services/user/User.service';
import { colors } from '../../../theme';
import { FontAwesome, SimpleLineIcons } from '@expo/vector-icons';
import { SideMenuService } from './SideMenu.service';
import Typography from '../typography/Typography';
import { setTestInfo } from '../../utils/test.utils';

export class Menu extends React.Component {
  onPress = async () => {
    SideMenuService.closeMenu();
    await AuthService.logout();
    await UserService.setUser(null);
    this.props.navigation.navigate('Login');
  };

  render() {
    return (
      <View
        style={{
          flex: 1,
          alignItems: 'stretch',
          backgroundColor: colors.card
        }}
      >
        <View
          style={{
            alignItems: 'center',
            borderBottomWidth: 1,
            paddingVertical: 20,
            marginHorizontal: 20,
            borderBottomColor: colors.border
          }}
        >
          <FontAwesome name="user-circle" size={120} color={colors.secondary} style={{ paddingBottom: 20 }} />
          <Typography color="secondary" size="small">
            {this.props.user && this.props.user.email}
          </Typography>
        </View>
        <View style={{ flex: 1, marginVertical: 10, width: 200, alignSelf: 'center', justifyContent: 'flex-end' }}>
          <RoundedButton
            startIcon={<SimpleLineIcons name="logout" size={24} color="white" />}
            label="Logout"
            onPress={this.onPress}
            {...setTestInfo('testTouchableLogout')}
          />
        </View>
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    user: state.user.data
  };
};

export default connect(mapStateToProps, null)(Menu);
