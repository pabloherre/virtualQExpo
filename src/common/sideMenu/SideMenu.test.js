import React from 'react';
import { Menu } from './SideMenu';
import { SideMenuService } from './SideMenu.service';
import UserService from '../../services/user/User.service';
import AuthService from '../../modules/auth/Auth.service';
import { shallow } from 'enzyme';

const mockProps = {
  navigation: {
    navigate: jest.fn()
  },
  user: {
    email: 'test@email'
  }
};

describe('<Menu />', () => {
  afterAll(() => {
    SideMenuService.closeMenu.mockClear();
    UserService.setUser.mockClear();
    AuthService.logout.mockClear();
    mockProps.navigation.navigate.mockClear();
  });

  it('renders correctly', async () => {
    const button = shallow(<Menu />);
    expect(button).toMatchSnapshot();
  });

  it('should logout on press button', async () => {
    const wrapper = shallow(<Menu {...mockProps} />);

    await wrapper.find("[testID='testTouchableLogout']").first().props().onPress();

    expect(SideMenuService.closeMenu).toHaveBeenCalled();
    expect(UserService.setUser).toHaveBeenCalled();
    expect(AuthService.logout).toHaveBeenCalled();
    expect(mockProps.navigation.navigate).toHaveBeenCalled();
  });
});
