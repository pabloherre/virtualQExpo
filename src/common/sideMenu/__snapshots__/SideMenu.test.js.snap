// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#e7eefb",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgb(216, 216, 216)",
        "borderBottomWidth": 1,
        "marginHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#0db4e4"
      name="user-circle"
      size={120}
      style={
        Object {
          "paddingBottom": 20,
        }
      }
    />
    <Component
      color="secondary"
      size="small"
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "flex-end",
        "marginVertical": 10,
        "width": 200,
      }
    }
  >
    <RoundedButton
      accessibilityLabel="testTouchableLogout"
      buttonStyle={Object {}}
      label="Logout"
      onPress={[Function]}
      startIcon={
        <Icon
          allowFontScaling={false}
          color="white"
          name="logout"
          size={24}
        />
      }
      testID="testTouchableLogout"
      textStyle={Object {}}
    />
  </View>
</View>
`;
